### DISCUSSION TITLE
Schemas for the data model used in PyU4V

### DISCUSSION THREAD

**Discussion by arthurzenika** [dell/PyU4V#164] [General]
It would be nice to have some graphical documentation of how the data model works in PyU4V and Dell's products it talks to. 

Here is a draft documentation I made (plantUML rendered by mermaidjs) : 

```mermaid
classDiagram
    class storage_group{
        StorageGroupId
        service_level
        slo_compliance
        type [Parent|Standalone|Child]
    }
    class masking_view{
        volumeId
    }
    class host{

    }
    masking_view --|> host
    masking_view --|> host_group
    host_group --|> host
    masking_view --|> port_group
    storage_group <|-- masking_view : maskingview
    storage_group <|-- storage_group: child_storage_group
    storage_group --|> storage
    storage --|> volume
    class storage{
        volumeId
        wwn
    }
    class volume{
        volumeId
        type
        emulation
        ssid
        allocated_percent
        cap_gb
        cap_mb
        cap_cyl
        status
        reserved
        pinned
        physical_name
        volume_identifier
        wwn
        encapsulated
        num_of_storage_groups
        num_of_front_end_paths
        pmaxname
        pmaxsymmetrixid
    }
    volume --|> storage_group: get_storage_group_from_volume(volumeId)
```

Do you think this could be useful to others ?

**Comment by rawstorage** [comment]
Hi Arthur, sorry for the late response on this one,  this is definitely useful however might be difficult to keep up to date.   I'll see if there is a way we can put a wiki on the Git or if we could include as part of the readthedocs.

**Comment by rawstorage** [comment]
Converted issue to discussion,  If you want to share any of these UML file with other customer you can fork the main branch of code and create a new folder UML,  create a pull request with the UML files you are creating and this will be visible.  Do you think this is acceptable?  Others may also share their UML.  I checked out our Git for some reason we don't have wiki function.
